controllers/controllers.go

```diff
@@ -167,6 +168,12 @@ func (r *instanceReconciler) deconfigureInstance(ctx context.Context, node *core
 	if err = nc.Deconfigure(ctx); err != nil {
 		return err
 	}
+
+	// Clean up node-specific RBAC after WICD cleanup is complete, before node deletion
+	if err = wmcorbac.CleanupNodeSpecificRBAC(ctx, r.client, r.k8sclientset, r.watchNamespace, instance.Node.GetName()); err != nil {
```

Comment:
If a node is force-removed or crashes, RBAC resources may remain. The cleanup function attempts to fix this, but isn't fail-proof. Consider a mechanism to clean up orphaned RBAC resources.